File: app/src/main/java/com/example/olesya/boardgames/ui/view/MainFragment.java
package com.example.olesya.boardgames.ui.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.example.olesya.boardgames.MenuFragment;
import com.example.olesya.boardgames.R;

public class MainFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_main, container, false);
        Button b = v.findViewById(R.id.g_imaginarium);
        b.setOnClickListener(view -> ((MainActivity)getActivity()).openFragment(new MenuFragment()));
        return v;
    }
}

File: app/src/main/java/com/example/olesya/boardgames/ui/main/MainFragment.kt
package com.example.olesya.boardgames.ui.main

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import com.example.olesya.boardgames.R


class MainFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.fragment_main, container, false)
        val b = v.findViewById<Button>(R.id.g_imaginarium)
        b.setOnClickListener { (activity as MainActivity).openFragment(MenuFragment()) }
        return v
    }
}